api/controllers/FlairController.js

```diff
@@ -116,16 +123,27 @@ module.exports = {
       promises.push(Reddit.setUserFlair(refreshToken, req.user.name, newsvFlair, flairs.svex, "SVExchange"));
       promises.push(User.update({name: req.user.name}, {loggedFriendCodes: friend_codes}));
 
-      if (!blockReport && (identical_banned_fcs.length || banned_alts.length || flagged.length)) {
+      if (!blockReport && (users_with_matching_fcs.length !== 0 || alt_user_names.length !== 0 || flagged.length)) {
         var message = 'The user /u/' + req.user.name + ' set the following flairs:\n\n' + flairs.ptrades + '\n\n' + flairs.svex + '\n\n';
-        if (identical_banned_fcs.length) {
-          message += '**This flair contains a banned friend code: ' + identical_banned_fcs + '**\n\n';
-        } else if (flagged.length && similar_banned_fcs.length) {
-          message += '**This flair contains a friend code similar to the following banned friend code'  + (similar_banned_fcs.length > 1 ? 's: ' : ': ') +
-            similar_banned_fcs.join(', ') + '**\n\n';
+        if (users_with_matching_fcs.length !== 0) {
+          message += 'This flair contains a friend code that matches ' + '/u/' + alt_users_fc.join(', /u/') + '\'s friend code: ' + matching_friend_codes + '\n\n';
+          var altNote = "Alt of " + alt_users_fc;
+          promises.push(Usernotes.addUsernote(refreshToken, 'FlairHQ', 'pokemontrades', req.user.name, altNote, 'spamwarn', ''));
+          var otherAltNote = "Alt of" + req.user.name;
+          promises.push(Usernotes.addUsernote(refreshToken, 'FlairHQ', 'pokemontrades', alt_users_fc, otherAltNote, 'spamwarn', ''));
+          if (identical_banned_fcs.length) {
+            message += '**' + identical_banned_fcs + ' is a banned friend code.**\n\n';
+          } else if (flagged.length && similar_banned_fcs.length) {
+            message += '**'  + (similar_banned_fcs.length > 1 ? 's: ' : ': ') + similar_banned_fcs.join(', ') + ' is similar to a banned friend code.**\n\n';
+          }
         }
-        if (banned_alts.length) {
-          message += '**This user may be an alt of the banned user' + (banned_alts.length === 1 ? '' : 's') + ' /u/' + banned_alts.join(', /u/') + '.**\n\n';
+        if (alt_user_names.length !== 0) {
+          message += 'This user may be an alt of the user' + (alt_user_names.length === 1 ? '' : 's') + ' /u/' + alt_user_names.join(', /u/') + '.\n\n';
+          promises.push(Usernotes.addUsernote(refreshToken, 'FlairHQ', 'pokemontrades', req.user.name, altNote, 'spamwarn', ''));
+          promises.push(Usernotes.addUsernote(refreshToken, 'FlairHQ', 'pokemontrades', alt_users_fc, otherAltNote, 'spamwarn', ''));
+          if (matching_banned_users) {
+            message += '**' + '/u/' + alt_user_names.join(', /u/') + ' is banned.**\n\n';
```

Comment:
This should be `matching_banned_users`, not `alt_user_names`. Otherwise if there are two alts and only one of them is banned, the message will say that both alts are banned.